phase2.js

```diff
@@ -1,62 +1,44 @@
-    //  'https://advantedgeadvisors.com/2017/07/18/3-ways-to-make-your-company-more-valuable-than-your-industry-peers/',
-    //  'https://advantedgeadvisors.com/2017/06/27/preparation-the-difference-between-successful-and-failed-deals/',
-    //  'https://advantedgeadvisors.com/2017/05/17/increasing-business-value-before-a-sale/'
-      'https://advantedgeadvisors.com/2017/04/24/how-to-avoid-7-common-deal-killers/',
-     'https://advantedgeadvisors.com/2017/03/21/the-benefits-of-an-ma-advisor/',
-      'https://advantedgeadvisors.com/2017/02/22/what-is-my-business-worth/'
-  ];
-  
+const fs = require('fs')
+
+const urls = fs.readFileSync('urls.txt', 'utf8').split('\n');
 const puppeteer = require('puppeteer');
-const list = hrefs.map ((href) => {
 
 (async () => {
   const browser = await puppeteer.launch();
   const page = await browser.newPage();
 
-    await page.goto(href, {
-        waitUntil: 'networkidle2',
+  const articles = [];
+
+  for (const url of urls) {
+    console.log(`Scraping ${url}`); 
+    await page.goto(url, { waitUntil: 'networkidle2' });
+    try { 
+      const article = await page.evaluate(()=> {
+        let background;
+        const backgroundImageElement = document.querySelector('.post > .elementor-section-wrap:first-child > .elementor-section:first-child')
+        if (backgroundImageElement) {
+          const style = getComputedStyle(backgroundImageElement)
+          const backgroundImage = style.backgroundImage || ''
+          const matches = /url\("([^"]*)"\)/.exec(backgroundImage)
+          background = matches && matches[1]
+        }
+        
+        const title = document.querySelector('.elementor-heading-title').innerText
+        const text = document.querySelectorAll('.elementor-text-editor.elementor-clearfix')
+
+        const body = Array.from(text).slice(0, -8).flatMap(node => node.innerHTML).join('\n');
+
+        return { background, title, body }
       });
-      let data = await page.evaluate(()=> {
-          const background = document.querySelector('style[id="elementor-frontend-inline-css"]').innerText.split('"')[1];
-          const title = document.querySelector('div[class="elementor-widget-container"] > h1').innerText;
-          const text = document.querySelectorAll('div[class="elementor-text-editor elementor-clearfix"] > p');
-
-          const allTexts = Array.from(text);
-          const textList = allTexts.map( a => a.innerText).toString();
-
-          const articles = {background: background, title: title, textList: textList};
-          const jsonArticles = JSON.stringify(articles);
-          return { jsonArticles }
-      })
-      console.log(data)
-    
-      await browser.close();
 
+      console.log({ url, ...article })
+      articles.push({ url, ...article })
+    } catch (e) {
+      console.error('Error scraping page.', e);
+    }
+  }
+
+  fs.writeFileSync('articles.json', JSON.stringify(articles, null, 2), 'utf8');

```

Comment:
Finally, we can write out the results to a JSON file. We use `JSON.stringify(articles, null, 2)`... the `, 2` part pretty-prints the JSON.